File: app/src/main/java/com/example/kolomachinetest/api/repo/marvel/data/Url.java
package com.example.kolomachinetest.api.repo.marvel.data;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class Url implements Parcelable {

    @SerializedName("available")
    public String type;

    @SerializedName("collectionURI")
    public String url;

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.type);
        dest.writeString(this.url);
    }

    public void readFromParcel(Parcel source) {
        this.type = source.readString();
        this.url = source.readString();
    }

    public Url() {
    }

    protected Url(Parcel in) {
        this.type = in.readString();
        this.url = in.readString();
    }

    public static final Parcelable.Creator<Url> CREATOR = new Parcelable.Creator<Url>() {
        @Override
        public Url createFromParcel(Parcel source) {
            return new Url(source);
        }

        @Override
        public Url[] newArray(int size) {
            return new Url[size];
        }
    };
}

File: app/src/main/java/com/example/kolomachinetest/api/repo/marvel/data/Url.kt
package com.example.kolomachinetest.api.repo.marvel.data

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import android.os.Parcel

class Url : Parcelable {
    @SerializedName("available")
    var type: String? = null

    @SerializedName("collectionURI")
    var url: String? = null
    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(type)
        dest.writeString(url)
    }

    fun readFromParcel(source: Parcel) {
        type = source.readString()
        url = source.readString()
    }

    constructor() {}
    protected constructor(`in`: Parcel) {
        type = `in`.readString()
        url = `in`.readString()
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Url> = object : Parcelable.Creator<Url> {
            override fun createFromParcel(source: Parcel): Url {
                return Url(source)
            }

            override fun newArray(size: Int): Array<Url?> {
                return arrayOfNulls(size)
            }
        }
    }
}